### PR TITLE
Test scripts for Education screen

### PR DESCRIPTION
**Pull request by chinthikaOHRM**
Test scripts for Education screen

### CODE REVIEW

File: symfony/test/functional/cypress/fixtures/user.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
     "admin": {
         "userName": "Admin",
-        "password": "Admin@123",
-        "fullName": "Samantha Jayasinghe"
+        "password": "admin@OHRM123",
+        "fullName": "Chinthika Rathnayake"
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/fixtures/user.json:5]: {your-admin-pwd-goes-here}

File: symfony/test/functional/cypress.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "baseUrl": "http://php80/orangehrm/symfony/web/index.php",
+    "baseUrl": "https://os50al9-os-guardian.orangehrm.com/web/index.php",
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress.json:2]: {your-base-url-goes-here}

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+//Get the current education list count
+const getEducationListCount = async() =>{ 
+    let txt = await promisify(cy.get('.oxd-text').contains('Found').invoke('text'))
+    return parseInt(txt.split(' ')[0])
+}
+
+describe('Qualification - Education', function () {
+
+    const characterLengthof51 = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:14]: better to define more than 50,100,200 chars in fixtures files

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    it('Add education',async () =>{
+        cy.login(user.admin.userName, user.admin.password)
+        cy.visit('/admin/viewEducation')
+        cy.wait(2000)
+
+
+        const previousCount = await getEducationListCount()
+        cy.log(previousCount)
+
+
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:45]: extra space

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        cy.get('.oxd-text--span').should('include.text', "Found")
+        cy.get('.oxd-button--medium').should('include.text',"Add").click()
+        cy.get('.oxd-input-group').type(educationName)
+        cy.get('.oxd-button--secondary').click()
+        cy.wait(500)
+        cy.get('.oxd-toast').should('include.text', 'Success')
+
+        cy.wait(7000)
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:53]: need to remove all waits

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 3 above)
Comment (chinthikaOHRM) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js]: I have defined different length chars in a fixtures file

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 4 above)
Comment (chinthikaOHRM) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js]: I have removed extra spaces

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 5 above)
Comment (chinthikaOHRM) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js]: Done

File: .vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (samanthajayasinghe) [inline .vscode/launch.json:1]: don't commit .vscode files

File: symfony/test/functional/.gitignore [hunk 7]
```diff
@@ -0,0 +1,5 @@
+node_modules
+*.log
+cypress/screenshots
+cypress/fixtures/user.json
```
Comment (samanthajayasinghe) [inline symfony/test/functional/.gitignore:4]: what if make `cypress/fixtures/admin_user.json` file add into git ignore

File: symfony/test/functional/cypress/fixtures/charLength.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
+{
+    "chars50": {
+        "level": "Aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/fixtures/charLength.json:3]: why it's 'level'? char50.text make sense to me

File: symfony/test/functional/cypress/fixtures/user.json (hunk 1 above)
Comment (samanthajayasinghe) [reply symfony/test/functional/cypress/fixtures/user.json:5]: we can above admin user in to `admin_user.json` file

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    let txt = await promisify(cy.get('.oxd-text').contains('Found').invoke('text'))
+    return parseInt(txt.split(' ')[0])
+}
+
+describe('Qualification - Education', function () {
+
+    const educationName = "Qqqqqqq"
+    const updatedEducationName = "Wwqqqqq"
+
+
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:18]: Extra spaces

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+//Get the current education list count
+const getEducationListCount = async () => {
+    let txt = await promisify(cy.get('.oxd-text').contains('Found').invoke('text'))
+    return parseInt(txt.split(' ')[0])
+}
+
+describe('Qualification - Education', function () {
+
+    const educationName = "Qqqqqqq"
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:15]: is it not good to have meaningful name?

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+        cy.login(user.admin.userName, user.admin.password)
+        cy.visit('/admin/viewEducation')
+        cy.get('.oxd-button--medium').should('include.text', "Add").click()
+        cy.get('.oxd-input-group').type(charLength.chars100.level)
+        cy.get('.oxd-input-group__message').should('include.text', 'Should not exceed 50 characters')
+        cy.get('.oxd-button--secondary').click()
+        cy.url().should('include', '/admin/saveEducation')
+    })
+
+
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:71]: pls format the code .. hard to read with extra white space lines :(

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+    //Validation check for the duplicated records
+
+    it('Duplicate Records Validation', () => {
+        cy.login(user.admin.userName, user.admin.password)
+        cy.login(user.admin.userName, user.admin.password)
+        cy.visit('/admin/viewEducation')
+        // cy.get('.oxd-button--medium').should('include.text',"Add").click()
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:80]: don't comment the code

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+        // cy.get('.oxd-toast').should('include.text', 'Success')
+        cy.get('.oxd-button--medium').should('include.text', "Add").click()
+        cy.get('.oxd-input-group').type(educationName)
+        cy.get('.oxd-button--secondary').click()
+        cy.get('.oxd-input-group__message').should('include.text', 'Qualification name should be unique')
+        cy.get('.oxd-button--secondary').click()
+        cy.url().should('include', '/admin/saveEducation')
+    })
+
+
```
Comment (samanthajayasinghe) [inline symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:92]: same here

**Comment by RajithaKumara** [comment]
Closing since open #842
@ShenaliFernando please fix the mentioned issued in this PR

File: symfony/test/functional/cypress/fixtures/user.json (hunk 1 above)
Comment (ShenaliFernando) [reply symfony/test/functional/cypress/fixtures/user.json]: Moved admin credentials to the admin.json file.

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 12 above)
Comment (ShenaliFernando) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js]: Removed commented code lines.

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 10 above)
Comment (ShenaliFernando) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js]: Added meaningful names.

File: symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js (hunk 9 above)
Comment (ShenaliFernando) [reply symfony/test/functional/cypress/integration/orangehrm/admin/education.spec.js:14]: Extra spaces removed.

**Pull request closed without merging**